### PR TITLE
Add library telemetry configuration consistency tests

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
## Description

Add test case to ensure consistency across APM client library configuration settings in both environment variables and reported APM telemetry. The telemetry data will power a new UI in the app that will let customers view the tracing configuration for their services.

image

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

https://datadoghq.atlassian.net/browse/APMON-305

### CODE REVIEW

**Review by avivenzio-dd — APPROVED** [review]

File: tests/parametric/test_telemetry.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    def test_library_settings(self, library_env, test_agent, test_library):
+        with test_library.start_span("test"):
+            pass
+        event = test_agent.wait_for_telemetry_event("app-started")
+        configuration = event["payload"]["configuration"]
+
+        configuration_by_name = {item["name"]: item for item in configuration}
+        for (apm_telemetry_name, value) in [
+            ("trace_sample_rate", "1.0"),
+            ("logs_injection_enabled", "false"),
```
Comment (ahmed-mez) [inline tests/parametric/test_telemetry.py:29]: How can we exclude unsupported configs? For instance logs injection in the Go tracer.

Comment (Kyle-Verhoog) [reply tests/parametric/test_telemetry.py:29]: Great point! I wonder if we should add an "unsupported" option to the telemetry data. I will follow up with the telemetry team about this.

Comment (Kyle-Verhoog) [reply tests/parametric/test_telemetry.py:29]: @mehulsonowal do you think it makes sense to have an explicit "unsupported" entry in the data for settings?

Comment (mehulsonowal) [reply tests/parametric/test_telemetry.py:29]: Hmm, so you mean have an additional field in the `client_library_configuration` table?

**Review by cbeauchesne — APPROVED** [review]

File: tests/parametric/test_telemetry.py (hunk 1 above)
Comment (smola) [reply tests/parametric/test_telemetry.py:29]: If it's unsupported, the entry should probably be missing?

**Review by smola — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]

**Pull request merged**